src/Json/SchemaLoader.php

```diff
@@ -61,7 +62,12 @@ public function load(string $file): stdClass
         $locatedFile = $this->fileLocator->locate($file);
 
         if (isset($this->schemas[$locatedFile]) === false) {
-            $schema = $this->dereferencer->dereference('file://'.$locatedFile);
+            if (in_array(pathinfo($locatedFile, PATHINFO_EXTENSION), array('yml', 'yaml'))) {
+                $dereference = (object) Yaml::parseFile($locatedFile);
+            } else {
+                $dereference = "file://{$locatedFile}";
```

Comment:
perhaps a check to determine that this file has json extension....sorta like you did with the yaml check

if not then throw an exception as your default clause.

Reply:
If not a `.yaml` or `.yml` extension, handling falls back to the same as it was ... if it's not JSON, an error should be generated

Reply:
```shell
$ php bin/console debug:router

In FileLoader.php line 168:
                                                                                                                                                                                                                                      
  Invalid JSON: Syntax error in ../src/Resources/api/v1/foo.txt (which is being imported from "/path/to/routes.yaml"). Make sure there is a loader supporting the "openapi" type.                                                                                                                                                                                                                           
                                                                                                                                                                                                                                      
In JsonDecoder.php line 46:
                              
  Invalid JSON: Syntax error 
```